src/components/inline_edit/inline_edit_form.tsx

```diff
@@ -147,15 +158,13 @@ export const EuiInlineEditForm: FunctionComponent<EuiInlineEditFormProps> = ({
   };
 
   const saveInlineEditValue = () => {
-    if (editModeValue && onConfirm && !onConfirm()) {
-      // If there is text, an onConfirm method is present, and it has returned false, cancel the action
+    if (onConfirm && !onConfirm(editModeValue)) {
+      // If an onConfirm method is present, and it has returned false, cancel the action
       return;
-    } else if (editModeValue) {
+    } else {
       setReadModeValue(editModeValue);
       setIsEditing(!isEditing);
-    } else {
-      // If there's no text, cancel the action, reset the input text, and return to readMode
-      cancelInlineEdit();
+      onSave && onSave(editModeValue);
```

Comment:
syntax succinctness nit
```suggestion
      onSave?.(editModeValue);
```